src/Container.php

```diff
@@ -178,6 +209,28 @@ public function call($alias, array $args = [])
     }
 
     /**
+     * Determines if a definition is registered via a service provider.
+     *
+     * @param  string $alias
+     * @return boolean
+     */
+    protected function providedByServiceProvider($alias)
+    {
+        foreach ($this->providers as $provider) {
+            $provider = ($provider instanceof ServiceProvider) ? $provider : new $provider;
```

Comment:
Good, don't let the ctor have arguments. There must not be anything injected in there as not everything is registered to the container yet.